tests/test_resources/test_modules/test_folder.py

```diff
@@ -183,49 +147,25 @@ def test_cluster_and_gcs(self, cluster, cluster_folder):
                 f"See https://cloud.google.com/sdk/gcloud/reference/auth/login"
             )
 
-    @pytest.mark.level("minimal")  # Needs S3 credentials
-    def test_s3_and_s3(self, local_folder, s3_folder):
-        # from one s3 folder to another s3 folder
-        new_s3_folder = s3_folder.to(system="s3")
-        assert "sample_file_0.txt" in new_s3_folder.ls(full_paths=False)
+    @pytest.mark.level("unit")
+    def test_send_folder_to_dest(self, folder, dest, cluster, tmp_path):
 
-        new_s3_folder.rm()
+        folder_mappings = {
+            "local_folder": {"system": "here", "_fs_str": "file"},
+            "local_folder_docker": {"system": "docker_cluster_pk_ssh_no_auth", "_fs_str": "ssh"},
+            "cluster_folder": {"system": "ondemand_cpu_cluster", "_fs_str": "ssh"},
+            "s3_folder": {"system": "s3", "_fs_str": "s3"},
+            "gcs_folder": {"system": "gs", "_fs_str": "gs"}
+        }
 
-    @pytest.mark.level(
-        "minimal"
-    )  # TODO: fix this test (Needs google-cloud-storage package)
-    @pytest.mark.skip("[WIP] Fix this test")
-    def test_gcs_and_gcs(self, gcs_folder):
-        # from one gcs folder to another gcs folder
-        new_gcs_folder = gcs_folder.to(system="gs")
-        assert "sample_file_0.txt" in new_gcs_folder.ls(full_paths=False)
-
-        new_gcs_folder.rm()
-
-    @pytest.mark.level("minimal")  # TODO: fix this test
-    @pytest.mark.skip("Doesn't work properly as only full-bucket copy is supported")
-    def test_s3_and_gcs(self, s3_folder):
-        # *** NOTE: transfers between providers are only supported at the bucket level at the moment (not directory) ***
-
-        s3_folder_to_gcs = s3_folder.to(system="gs")
-        assert "sample_file_0.txt" in s3_folder_to_gcs.ls(full_paths=False)
-        assert s3_folder_to_gcs._fs_str == "gs"
+        new_folder = folder.to(system=folder_mappings[dest]["system"])
```

Comment:
this can probably be simplified to something like:

```
if instance(dest.system, cluster):
    expected_fs_str = "file" if dest.system == self.system else "ssh"
else:
    expected_fs_str = dest.system

new_folder = folder.to(system=dest.system)
assert new_folder._fs_str == expected_fs_str
assert ...
```